File: 0067-add-binary/0067-add-binary.cpp
class Solution {
public:
    string addBinary(string a, string b) {
        int i = a.size()-1;
        int j = b.size()-1;
        int remember = 0;
        int temp;
        string result = "";
        while (i>=0||j>=0||remember!=0){
            temp = remember;
            if (i>=0){
                temp += a[i] - '0';
                i--;
            }
            if (j>=0){
                temp += b[j] - '0';
                j--;
            }
            result += temp % 2 + '0';
            remember = temp / 2;
        }
        reverse(result.begin(), result.end());
        return result;
    }
};

File: 0067-add-binary/0067-add-binary.py
class Solution:
    def addBinary(self, a: str, b: str) -> str:
        i, j = len(a) - 1, len(b) - 1
        res = []
        remember = 0
        while i >= 0 or j >= 0 or remember > 0:
            temp = remember
            if i >= 0:
                temp += int(a[i])
                i -= 1
            if j >= 0:
                temp += int(b[j])
                j -= 1
            res.append(str(temp % 2))
            remember = temp // 2
        return "".join(res[::-1])
